### PR TITLE
Explicitly set `readOnlyRootFilesystem: false` on created registry pods.

### PR DESCRIPTION
**Pull request by joelanford**
When running in environments that impose strict security (e.g.
OpenShift with unknown SCC defaults), it is important to be explicit
about the security context requirements of the pods we run to avoid
runtime failures due to missing permissions.

This commit explicitly sets `readOnlyRootFilesystem: false` because
the registry pods need to copy the registry pod's databile file to
a separate read-only copy.

Signed-off-by: Joe Lanford <joe.lanford@gmail.com>

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [x] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [x] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/controller/registry/reconciler/reconciler.go
Comment (timflannagan) [inline pkg/controller/registry/reconciler/reconciler.go:7]: I'd like to see us standardize these package import names and have all of these be `corev1` at some point.

I'll briefly hold in case someone else wants to review but these changes sound reasonable to me.

/lgtm
/hold

**Review by ankitathomas — APPROVED** [review]
/lgtm

**Comment by timflannagan** [comment]
JK, I realized we already have an approval

/hold cancel

File: pkg/controller/registry/reconciler/reconciler.go
Comment (joelanford) [reply pkg/controller/registry/reconciler/reconciler.go:7]: I blindly use https://github.com/AanZee/goimportssort as a file watcher in Goland, so I don't pay much attention to the imports block. It probably wouldn't be too hard to write a simple check that looks for some known api packages and checks that the import name follows that pattern.

**Pull request merged**
